src/morse/polling.clj

```diff
@@ -21,22 +21,34 @@
    Returns channel with updates from Telegram"
   [running token opts]
   (let [updates (a/chan)
+        updates-caller (or (:get-updates-fn opts) api/get-updates)
         timeout (or (:timeout opts) 1000)]
     (go-loop [offset 0]
-      (let [response (a/thread (api/get-updates token (merge opts {:offset offset})))
-            [data _] (a/alts! [running response])]
+      (let [response (a/chan)
+            ;; in clj-http async handlers the context is
+            ;; outside of go-loop - that's why using >!! here
+            _ (updates-caller
+               token
+               #(>!! response %)
+               (fn [exception]
```

Comment:
this is kind of abstract and not really relevant to polling, so should be extracted